lib/replica_pools/connection_proxy.rb

```diff
@@ -109,7 +109,11 @@ def within_leader_block?
 
     def route_to(conn, method, *args, **keyword_args, &block)
       raise ReplicaPools::LeaderDisabled.new if ReplicaPools.config.disable_leader && conn == leader
-      conn.retrieve_connection.send(method, *args, **keyword_args, &block)
+      connection = conn.retrieve_connection
+
+      # in rails 7.1 we can use the verify! method for a cleaner implementation
```

Comment:
I don't see `verify!` being used here — out of date commend?

Reply:
In rails 7.0.8 (our current version)`verify!` does what I've put here, checks `.active?` then `.reconnect!` if false. But in 7.1 `verify!` will do that plus some other stuff and return true once the connection is good. There we could change this to be `connection.send(method, *args, **keyword_args, &block) if connection.verify!`